tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/ContextPropagation.cs

```diff
@@ -27,14 +28,34 @@ private static void Inject<TMessageRequest>(SpanContext context, IDictionary mes
             sb.Append('}');
 
             var resultString = Util.StringBuilderCache.GetStringAndRelease(sb);
-            byte[] bytes = Encoding.UTF8.GetBytes(resultString);
-            MemoryStream stream = new MemoryStream(bytes);
+            var bytes = Encoding.UTF8.GetBytes(resultString);
+            var stream = new MemoryStream(bytes);
             messageAttributes[SnsKey] = CachedMessageHeadersHelper<TMessageRequest>.CreateMessageAttributeValue(stream);
         }
 
-        public static void InjectHeadersIntoMessage<TMessageRequest>(IContainsMessageAttributes carrier, SpanContext spanContext)
+        public static void InjectHeadersIntoBatch<TBatchRequest>(IPublishBatchRequest request, SpanContext context)
         {
-            // add distributed tracing headers to the message
+            // Skip adding Trace Context if entries don't exist or empty.
+            if (request.PublishBatchRequestEntries is not { Count: > 0 })
+            {
+                return;
+            }
+
+            var publishBatchRequestEntry = request.PublishBatchRequestEntries[0].DuckCast<IContainsMessageAttributes>();
```

Comment:
Do we only want to inject into the _first_ entry? Not all entries? 🤔

Reply:
Hey – I had a chat with the Serverless APM team to talk about this, I think injections should be done on every Message. I've confirmed we do this for Java, but not for every other runtime. I'll move on and update this so we inject on every Message, thanks for bringing it up!

Reply:
I believe each entry would have its own corresponding request/trace right? so I _think_ we would want to inject on every entry but my confidence is not that high in this statement without a quick manual test